### ISSUE TITLE
[BUG] options parsing is broken in 2.6.0

### ISSUE THREAD

**Issue by dch** [meltwater/gen_rmq#177]
**Describe the bug**

On 2.5.0, the following is valid:

```
  def init() do
    [
      connection:
        "amqps://user:passwd@host:5671/vhost?heartbeat=37&verify=verify_peer&cacertfile=/usr/local/share/certs/ca-root-nss.crt&server_name_indication=*.com&fail_if_no_peer_cert=true&depth=3",
      queue: "api.fudge",
      queue_options: [durable: true],
      exchange: {:topic, "api.topic"},
      routing_key: "api.fudge",
      prefetch_count: "1",
      deadletter: false,
      retry_delay_function: fn attempt ->
        :timer.sleep(2000 * attempt - floor(1000 * :rand.uniform()))
      end
    ]
  end
```

**To Reproduce**

1. create consumer using above `init/1` form 2.6.0
2. watch it crash

```iex
14:44:25.105 [error] CRASH REPORT Process <0.302.0> with 1 neighbours crashed with reason: #{'__exception__' => true,'__struct__' => 'Elixir.Protocol.UndefinedError',description => <<>>,protocol => 'Elixir.Enumerable',value => nil} in 'Elixir.Enumerable':'impl_for!'/1 line 1
** (EXIT from #PID<0.300.0>) shell process exited with reason: an exception was raised:
    ** (Protocol.UndefinedError) protocol Enumerable not implemented for nil of type Atom. This protocol is implemented for the following type(s): File.Stream, HashDict, List, IO.Stream, GenEvent.Stream, Function, MapSet, Map, Stream, HashSet, Range, Date.Range
        (elixir 1.10.3) lib/enum.ex:1: Enumerable.impl_for!/1
        (elixir 1.10.3) lib/enum.ex:141: Enumerable.reduce/3
        (elixir 1.10.3) lib/enum.ex:959: Enum.find/3
        (gen_rmq 2.6.0) lib/gen_rmq/consumer/queue_configuration.ex:66: GenRMQ.Consumer.QueueConfiguration.remove_keyword/3
        (gen_rmq 2.6.0) lib/gen_rmq/consumer/queue_configuration.ex:60: GenRMQ.Consumer.QueueConfiguration.combine_options/2
        (gen_rmq 2.6.0) lib/gen_rmq/consumer/queue_configuration.ex:14: GenRMQ.Consumer.QueueConfiguration.setup/2
        (gen_rmq 2.6.0) lib/consumer.ex:373: GenRMQ.Consumer.parse_config/1
        (gen_rmq 2.6.0) lib/consumer.ex:259: GenRMQ.Consumer.init/1
```

**Environment**

tested at least with these:

- Elixir version:  1.10.2 & 1.10.3
- Erlang/OTP version: OTP22
- RabbitMQ version: 3.8.1 & 3.8.3

**Additional context**

**Comment by mkorszun** [comment]
@dch thanks for reporting. Fix is already waiting for review - https://github.com/meltwater/gen_rmq/pull/178

**Comment by mkorszun** [comment]
@dch patch version has been released - `2.6.1`. Looking for your feedback.
